### DISCUSSION TITLE
How to style GeoJSON layer according to embedded attributes

### DISCUSSION THREAD

**Discussion by rsurgiewicz** [maplibre/maplibre-gl-js#1084] [Q&A] [answered]
Hi Folks,

I have a question. The datasource I'm using provides the full geojson data with attributes e.g

```
        "fill-opacity": 0.5,
        "stroke-width": 0.5,
        "stroke": "#8f8f8f"
```
and it is interpreted correctly e.g by geojson.io
![image](https://user-images.githubusercontent.com/47541623/157638028-7406e0b2-a255-4e39-9772-8b0be35a648f.png)

How can I force `maplibre-gl` to use those styles, cause all I get is:
![image](https://user-images.githubusercontent.com/47541623/157638192-b42c04f6-a696-49b5-8441-05c832ecb0a0.png)

Thanks!

**Comment by xabbu42** [comment]
I assume you are talking about https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0? this is not part of the geojson spec so talking about "full geojson data" is a bit misleading. It should be quite easy to make a style using expressions to support most of this properties (not sure about the marker ones). This would also make a nice documentation example as this could be a common use case.

**Comment by rsurgiewicz** [comment, marked as answer]
Ok finally I got it right.
Actually there's a possibility to use `get` feature to grab value from data.

so in my example 
```
...      },
      "properties": {
        "fill": "#18ff00",
        "fill-opacity": 0.5,
        "stroke-width": 0.5,
        "stroke": "#8f8f8f"
      }
```
it would be something like:

```export const styleLayer = {
  id: "data",
  type: "fill",
  paint: {
    "fill-color": ["get", "fill"],
    "fill-opacity": 0.8,
  },
};
````

**Reply by lulumalik** [reply]
nice , it's help me a lot
